AprilSylph/XKit-Rewritten
# Tag Tracking+: Improve count resetting

### Description


Not a great fix for #912, as it doesn't address the fundamental race condition if sets of real posts were loaded close enough together, which I bet can still happen. But this quick-and dirty fixes the case I was observing with the ad posts.

Prevents a set of ads loading on a tracked tag page from overwriting the timestamp storage by only writing if there was a valid update.

### Testing steps

Kind of hard to test; my account isn't getting ads on tagged pages right now. In theory, though: disable ad-free, track a tag that you haven't seen before, refresh on the dashboard, then soft navigate to the newly followed tag using the search dropdown and see if the Tag tracking+ count zeroes even if ads appear late.
